node/src/eventsbuilder.ts

```diff
@@ -25,12 +28,21 @@ export class EventsBuilder {
 
     getStartPosition(): SeekPosition {
         const result = new SeekPosition();
-
+        const specified = new SeekSpecified();
         const startBlock = this.#options.startBlock;
+        const checkPointer = this.#options.checkpointer;
+
+        if (checkPointer) {
+            const blockNumber = checkPointer.getBlockNumber();
+            if (blockNumber != undefined) {
```

Comment:
`blockNumber` is the `startBlock`, and there is already code below to deal with the start block correctly so this is duplication. You might be able to replace this entire piece of checkpointer code this the following line above:
```typescript
const startBlock = this.#options.checkpointer?.getBlockNumber() ?? this.#options.startBlock;
```